### PR TITLE
Peterpur

### PR DESCRIPTION
**Pull request by peterpur**
review feedback & rejected challenges

### CODE REVIEW

File: Evaluation/views.py [hunk 1]
```diff
@@ -802,14 +810,22 @@ def sort(request, course_short_title=None):
 
 
 @login_required()
-@staff_member_required
 def get_points(request, user, course):
+    is_correct_user_request = RequestContext(request)['user'].id is user.id
+    is_staff_request = RequestContext(request)['user'].is_staff
+    if not (is_correct_user_request or is_staff_request):
+        raise Http404
```
Comment (queltos) [inline Evaluation/views.py]: 403 (Forbidden) wär hier passender als 404 (not found)

File: Review/admin.py [hunk 2]
```diff
@@ -56,7 +56,42 @@ class ReviewEvaluationAdmin(admin.ModelAdmin):
             }
         ),
     ]
-    list_display = ('id', 'review', 'creation_time', 'user', 'appraisal', )
+
+    def get_review_id(self, review_evaluation):
+        url = '<a href="/admin/Review/review/%s/">%s</a>'
+        review_id = review_evaluation.review.id
+        return url % (review_id, review_id)
```
Comment (queltos) [inline Review/admin.py]: This is the old way of string formatting. The new one looks something like this:
```python
'<a href="/admin/Review/review/{}/">{}</a>'.format(review_id, review_id)
```
You could also do something like
```python
"...{0}...{0}...".format(review_id)
```
or use named parameters.

See https://docs.python.org/3/library/string.html#string-formatting

File: templates/points.html [hunk 3]
```diff
@@ -1,42 +1,91 @@
...
                 <td class="stack_name">
-					<a style="text-decoration:none; color:black;" href="/{{course.short_title}}/challenge/stack?id={{stack.stack.id}}">
-						{% if stack.points_earned != 0 %} <i class="fa fa-check-circle" style="color:gray"></i> {% else %} <i style="color:#3AC958" class="fa fa-circle-o"></i>{% endif %}
-						{{ stack.stack.title }}</a></td>
-                <td class="punkte" style="vertical-align:bottom;">{{ stack.points_earned }}/{{ stack.points_available }}</td>
-            </tr>
+                  <a style="text-decoration:none; color:black;"
+                     href="/{{ course.short_title }}/challenge/stack?id={{ stack.stack.id }}">
+                    {% if stack.status.status == "Evaluated." %}
+						{% if stack.points_earned == 0 %}
```
Comment (queltos) [inline templates/points.html]: indentations?

Comment (peterpur) [inline templates/points.html]: looks fine in my editor. in other words: i indent with tabs, other people use spaces (which is a waste of bandwidth, in my eyes). mixing spaces + tabs can lead to somthing like this, if your default TAB-distance is different from mine. 




> On 15.5.2015, at 16:31, queltos <notifications@github.com> wrote:
> 
> In templates/points.html:
> 
>>                 <td class="stack_name">
>> -					<a style="text-decoration:none; color:black;" href="/{{course.short_title}}/challenge/stack?id={{stack.stack.id}}">
>> -						{% if stack.points_earned != 0 %} <i class="fa fa-check-circle" style="color:gray"></i> {% else %} <i style="color:#3AC958" class="fa fa-circle-o"></i>{% endif %}
>> -						{{ stack.stack.title }}</a></td>
>> -                <td class="punkte" style="vertical-align:bottom;">{{ stack.points_earned }}/{{ stack.points_available }}</td>
>> -            </tr>
>> +                  <a style="text-decoration:none; color:black;"
>> +                     href="/{{ course.short_title }}/challenge/stack?id={{ stack.stack.id }}">
>> +                    {% if stack.status.status == "Evaluated." %}
>> +						{% if stack.points_earned == 0 %}
> 
> indentations?
> 
> —
> Reply to this email directly or view it on GitHub.
> 



::DVR 0005886::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

I need someone to protect me from all the measures they take in order to protect 
me. [Banksy]

Comment (martflu) [inline templates/points.html]: I used pycharms auto formatting.

**Comment by martflu** [comment]
should be ready to merge now

**Pull request merged**